pycbc/psd/variation.py

```diff
@@ -210,10 +210,197 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
+        irfft(fwhiten * fweight), int(psd_duration / 2) * sample_rate)
+
+    return full_filt
+
+
+def live_calc_psd_variation(strain,
+                            full_filt,
+                            increment,
+                            data_trim=2.0,
+                            short_stride=0.25):
+    """
+    Calculate the psd variation in the PyCBC Live search.
+
+    The Live strain data is convolved with the filter to produce a timeseries
+    containing the PSD variation values for each sample. This mean square of
+    the timeseries is then taken over the short_stride to remove the effects of
+    short duration glitches and further outliers from the mean are replaced
+    within the array. This array is then further averaged every second to
+    produce a timeseries that will contain a number of value equal to the
+    increment.
+
+    Parameters
+    ----------
+    strain : pycbc.timeseries
+        Live data being searched through by the PyCBC Live search.
+    full_filt : some sort of array, scipy so i guess numpy?
+        A filter created by `live_create_filter`.
+    increment : float
+        The number of seconds in each increment in the PyCBC Live search.
+    data_trim : float
+        The number of seconds to be trimmed from either end of the convolved
+        timeseries to prevent artefacts.
+    short_stride : float
+        The number of seconds to average the PSD variation timeseries over to
+        remove the effects of short duration glitches.
+
+    Returns
+    -------
+    psd_var : pycbc.timeseries
+        A timeseries containing the PSD variation values for each second
+        an increments number of seconds prior to the end of the input strain
+        data.
+
+    """
+    sample_rate = int(strain.sample_rate)
+
+    # Grab the last increments worth of data with extra to account for the need
+    #  to trim the data to remove edge effects.
+    astrain = strain.time_slice(strain.end_time - increment - (data_trim*3),
+                                strain.end_time)
+
+    # Convole the data and the filter to produce the PSD variation timeseries,
+    #  then trim the beginning and end of the data to prevent edge effects.
+    wstrain = sig.fftconvolve(astrain, full_filt, mode='same')
+    wstrain = wstrain[int(data_trim * sample_rate):-int(data_trim * sample_rate)]
+
+    # Create a PSD variation array by taking the mean square of the PSD
+    #  variation timeseries every short_stride
+    short_ms = numpy.mean(
+        wstrain.reshape(-1, int(sample_rate * short_stride)) ** 2, axis=1)
+
+    # Create an array of averages to substitute out outliers in the PSD
+    #  variation array
+    ave = 0.5 * (short_ms[2:] + short_ms[:-2])
+    outliers = short_ms[1:-1] > (2. * ave)
+    short_ms[1:-1][outliers] = ave[outliers]
+
+    # Calculate the average of the PSD variation array for every second
+    m_s = []
+    stride = 1 / short_stride
```

Comment:
this is a bit confusing - 'short_stride' is a time duration in seconds, so the variable you are calling 'stride' is an inverse time, or frequency.  This whole block probably needs a bit more explanation as to what all the different variables and indices are doing though (the original Simone code is not very clear either)

Reply:
I have updated the variable 'stride' to be 'samples_per_second' to help indicated what the variable actually represents. We want to calculate how many elements in 'short_ms' form 1 second's worth of data and then take an average of those samples.

I've also updated the comment to help:
"    # Calculate the PSD variation every second by averaging over short_ms every
    #  samples_per_second."